### ISSUE TITLE
MPI_Status_set_elements_x() does not work with an -m32 build (v1.7)

### ISSUE THREAD

**Issue by ompiteam** [open-mpi/ompi#187]
When you build OMPI (trunk or v1.7) with:

```
./configure CFLAGS=-m32 CXXFLAGS=-m32 FCFLAGS=-m32 --with-wrapper-cflags=-m32 --with-wrapper-cxxflags=-m32 --with-wrapper-fcflags=-m32
```

you end up with MPI_Count being a long long, which is 8 bytes.  But size_t is 4 bytes.

When you call MPI_Status_set_elements_x() with a value larger than 2^32^ (e.g., the ibm datatypes/getel_x test), it gets passed in to MPI_Status_set_elements_x() properly, but then it calls ompi_datatype_set_element_count(), which passes the MPI_Count value through a parameter that is of type size_t, and the value gets truncated.

Hence, the value that is set on the status is the truncated value, not the actual larger-than-2^32^ value.

This is the v1.7 version of https://svn.open-mpi.org/trac/ompi/ticket/4205.  Due to ABI issues, the solution for v1.7/v1.8 will be different than the solution for trunk/v1.9.

**Comment by jsquyres** [comment]
This was fixed on the SVN trunk in Feb of 2014.  It's therefore already included in the v2.x branch.

Since this apparently won't get fixed for v1.8 / v1.10, we can just closed this issue.

**Comment by jsquyres** [comment]
Finally closing the issue on 13 Oct 2015.
